taxi/views.py

```diff
@@ -104,16 +135,12 @@ class DriverLicenseUpdateView(LoginRequiredMixin, generic.UpdateView):
 
 class DriverDeleteView(LoginRequiredMixin, generic.DeleteView):
     model = Driver
-    success_url = reverse_lazy("")
+    success_url = reverse_lazy("taxi:driver-list")
 
 
-@login_required
+# Function to toggle the assignment of a car
 def toggle_assign_to_car(request, pk):
-    driver = Driver.objects.get(id=request.user.id)
-    if (
-        Car.objects.get(id=pk) in driver.cars.all()
-    ):  # probably could check if car exists
-        driver.cars.remove(pk)
-    else:
-        driver.cars.add(pk)
-    return HttpResponseRedirect(reverse_lazy("taxi:car-detail", args=[pk]))
+    car = Car.objects.get(pk=pk)
+    car.is_assigned = not car.is_assigned  # Toggle the assignment status
```

Comment:
The `Car` model does not have an `is_assigned` field. You need to add a Boolean field `is_assigned` to the `Car` model or modify this function to use an existing field.